Linter times out after 5m in CI

/kind bug

**What steps did you take and what happened:**

In the last few weeks, we've seen intermittent failures of the basic `test` CI job on the `make lint` step. Here's [one example](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/3105/pull-cluster-api-provider-azure-test/1618767709313437696).

```
level=info msg="[runner] Processors filtering stat (out/in): path_prettifier: 1407/1407, skip_dirs: 960/960, filename_unadjuster: 1407/1407, skip_files: 960/1407, autogenerated_exclude: 842/960, exclude: 842/842, exclude-rules: 33/842, nolint: 0/33, cgo: 1407/1407, identifier_marker: 842/842"
level=info msg="[runner] processing took 105.42907ms with stages: exclude-rules: 39.989853ms, identifier_marker: 31.4026ms, autogenerated_exclude: 11.493946ms, path_prettifier: 10.317436ms, nolint: 9.746662ms, skip_dirs: 1.49751ms, skip_files: 525.256µs, cgo: 285.647µs, filename_unadjuster: 160.866µs, max_same_issues: 2.24µs, uniq_by_line: 2.233µs, exclude: 1.39µs, path_shortener: 807ns, diff: 598ns, sort_results: 513ns, source_code: 398ns, severity-rules: 346ns, max_per_file_from_linter: 327ns, max_from_linter: 306ns, path_prefixer: 136ns"
level=info msg="[runner] linters took 3m3.43775341s with stages: goanalysis_metalinter: 3m3.33125089s, rowserrcheck: 163.043µs"
level=info msg="File cache stats: 560 entries of total size 4.5MiB"
level=error msg="Timeout exceeded: try increasing it by passing --timeout option"
make: *** [Makefile:529: lint] Error 4
```

**What did you expect to happen:**

Linter should complete and only error if actual linting issues are found.

**Anything else you would like to add:**


**Environment:**

This doesn't happen for me locally, only sometimes in our `prow` CI environment.

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -1,5 +1,5 @@
 run:
-  deadline: 5m
+  deadline: 10m
   skip-files:
   - 'zz_generated\.(\w*)\.go$'
   build-tags:

